gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ValidateAPIController.java

```diff
@@ -69,33 +73,62 @@ public class ValidateAPIController {
     public ResponseEntity<String> checkConformance(@PathVariable String serviceId) {
         ConformanceProblemsContainer foundNonConformanceIssues = new ConformanceProblemsContainer(serviceId);
         foundNonConformanceIssues.put(CONFORMANCE_PROBLEMS, validateServiceIdFormat(serviceId));
-
-        if (foundNonConformanceIssues.size() != 0) {
+        if (!foundNonConformanceIssues.isEmpty())
             return generateBadRequestResponseEntity(NON_CONFORMANT_KEY, foundNonConformanceIssues);
-        }
 
-        foundNonConformanceIssues.put(REGISTRATION_PROBLEMS, checkOnboarding(serviceId));
-        if (foundNonConformanceIssues.size() != 0) {     // cant continue if a service isn't registered
-            return generateBadRequestResponseEntity(WRONG_SERVICE_ID_KEY, foundNonConformanceIssues);
-        }
+        try {
+            checkOnboarding(serviceId);
 
-        List<ServiceInstance> serviceInstances = discoveryClient.getInstances(serviceId);
-        foundNonConformanceIssues.put(REGISTRATION_PROBLEMS, instanceCheck(serviceInstances));
-        if (foundNonConformanceIssues.size() != 0) {     // cant continue if we cant retrieve an instance
-            return generateBadRequestResponseEntity(WRONG_SERVICE_ID_KEY, foundNonConformanceIssues);
-        }
+            List<ServiceInstance> serviceInstances = discoveryClient.getInstances(serviceId);
+            instanceCheck(serviceInstances);
 
-        ServiceInstance serviceInstance = serviceInstances.get(0);
-        Map<String, String> metadata = getMetadata(serviceInstance);
+            ServiceInstance serviceInstance = serviceInstances.get(0);
```

Comment:
Don't we need to check if the serviceInstances is empty or not before accessing it?

Reply:
This is checked in the instanceCheck() function above

Reply:
Renamed the function so it should be clearer now